fetch/peers/peers.go

```diff
@@ -101,12 +101,14 @@ func (p *Peers) OnLatency(id peer.ID, size int, latency time.Duration) {
 	peer.success++
 	peer.failRate = float64(peer.failures) / float64(peer.success+peer.failures)
 	if peer.averageLatency != 0 {
-		peer.averageLatency = 0.8*peer.averageLatency + 0.2*float64(latency)
+		delta := (float64(latency) - float64(peer.averageLatency)) / 10 // 86% of the value is the last 19
+		peer.averageLatency += delta
```

Comment:
Could it make sense to use an exponential moving average here (and below with a different period size)?

```go
const k = 2/(n+1) // where n is the period size (>= 2), larger n reduce the effects of spikes
peer.averageLatency = latency*k + peer.averageLatency*(1-k)
```

Reply:
Nvm, the current code is basically the same with `n == 19` 🤦

Reply:
what i want to try is to move faster away from a node that was fast, and then became slow or faulty.
i looked up solution in lotus and seems to do exactly what i wanted, i will get back to this if it proves insufficient